pages/titles/[id].js

```diff
@@ -0,0 +1,38 @@
+import { data } from "../../utils/data"
+
+// export paths
+export const getStaticPaths = async() => {
+  const paths = data.map(item => {
+    return {
+      params: { id: item.id.toString() }
+    }
+  })
+  return {
+    paths,
+    fallback: false
+  }
+}
+
+// pass props to TitleDetail
+export const getStaticProps = async (context) => {
+  const id = context.params.id
+  return {
+    props: { id: id },
+   }
+}
+
+function TitleDetail ({ id }) {
+  const oneDetail = data.find((item) => {
```

Comment:
1. Please use `useMemo`
2. `oneDetail` looks like a weird name, `detail` will be better.